contents/java/JavaReflections.md

```diff
@@ -0,0 +1,291 @@
+# Advanced Java: Reflections
+
+Authors: [Jeremy Goh](https://github.com/MightyCupcakes)
+
+## Reflection
+
+### What is Java reflections
+
+Java reflections allow the inspection of classes, methods and fields during runtime, without having any knowledge of it
+during compile time. It is also possible to create new instance of existing objects, invoke methods and change values 
+of fields of existing objects. Hence the name reflection because Java is in a way looking at itself during runtime!
```

Comment:
Suggest explaining what reflection is without referring to Java first, since it is a general feature that other languages support as well.